### ISSUE TITLE
function [nngen/basic_type.py:__Operator collect_numerics] is too busy

### ISSUE THREAD

**Issue by RyusukeYamano** [NNgen/nngen#38]
The function does not terminate in deep (& complex) networks because it re-explores the already explored ”args” when exploring the computational graph.

https://github.com/NNgen/nngen/blob/2a30d83a040871f37a1d3bb17ac5e314d1b9d1b6/nngen/basic_types.py#L476-L484

## This seems to work well.

``` python
    def collect_numerics(self, ret=set()):
        ret.add(self)
        for arg in self.args:
            if arg not in ret:
                ret = ret.union(arg.collect_numerics(), ret)

        return ret
```

**Comment by shtaxxx** [comment]
Is this fine?

```python
    def collect_numerics(self, ret=set()):
        ret.add(self)
        for arg in self.args:
            if arg not in ret:
                ret = arg.collect_numerics(ret)
        return ret
```

**Comment by shtaxxx** [comment]
Fixed in 6d26c8cbfc68be6d5614827d1d390312bc630873
